testing/citest/tests/gcs_pubsub_gae_test.py

```diff
@@ -0,0 +1,453 @@
+    """Initialize command line argument parser.
+
+    Args:
+    parser: argparse.ArgumentParser
+    """
+    super(GcsPubsubGaeTestScenario, cls).initArgumentParser(
+      parser, defaults=defaults)
+
+    defaults = defaults or {}
+    parser.add_argument(
+      '--app_directory_root', default=None,
+      help='Path from the root of source code repository to the application directory.')
+    parser.add_argument(
+      '--branch', default='master',
+      help='Git branch to be used when deploying from source code repository.')
+    parser.add_argument(
+      '--git_repo_url', default=None,
+      help='URL of a git source code repository used by Spinnaker to deploy to App Engine.')
+    parser.add_argument(
+      '--test_gcs_bucket', default=None,
+      help='GCS bucket to upload GAE app source code to.')
+    parser.add_argument(
+      '--test_storage_account_name', default=None,
+      help='Storage account when testing GCS buckets.'
+      ' If not specified, use the application default credentials.')
+    parser.add_argument(
+      '--test_subscription_name', default=None,
+      help='Google pub/sub subscription name configured in Echo.')
+
+  def __init__(self, bindings, agent=None):
+    super(GcsPubsubGaeTestScenario, self).__init__(bindings, agent)
+    self.logger = logging.getLogger(__name__)
+
+    bindings = self.bindings
+
+    if not bindings['GIT_REPO_URL']:
+      raise ValueError('Must supply value for --git_repo_url')
+
+    if not bindings['APP_DIRECTORY_ROOT']:
+      raise ValueError('Must supply value for --app_directory_root')
+
+    # We'll call out the app name because it is widely used
+    # because it scopes the context of our activities.
+    self.TEST_APP = bindings['TEST_APP']
+    self.TEST_STACK = bindings['TEST_STACK']
+    self.EXPECTED_ARTIFACT_ID = 'deployable-gae-app-artifact'
+
+    self.__gcp_project = bindings['APPENGINE_PRIMARY_MANAGED_PROJECT_ID']
+    self.__cluster_name = frigga.Naming.cluster(self.TEST_APP, self.TEST_STACK)
+    self.__server_group_name = frigga.Naming.server_group(self.TEST_APP, self.TEST_STACK)
+    self.__lb_name = self.__cluster_name
+
+    self.__subscription_name = bindings['TEST_SUBSCRIPTION_NAME']
+    self.__gcs_pubsub_agent = sk.GcsPubsubAgent(bindings['APPENGINE_CREDENTIALS_PATH'])
+
+    # Python is clearly hard-coded as the runtime here, but we're just asking App Engine to be a static file server.
+    self.__app_yaml = ('\n'.join(['runtime: python27',
+                                  'api_version: 1',
+                                  'threadsafe: true',
+                                  'service: {service}',
+                                  'handlers:',
+                                  ' - url: /.*',
+                                  '   static_dir: .']).format(service=self.__lb_name))
+
+    self.__app_directory_root = bindings['APP_DIRECTORY_ROOT']
+    self.__branch = bindings['BRANCH']
+    self.temp = tempfile.mkdtemp()
```

Comment:
You create this directory here in the scenario constructor, but then delete it in the fixture tear down. That's violating ownership. The scenario can span fixtures. It doesnt now, but doing this makes it less maintainable.

Either you should create the directory in the fixture setUpClass method and inject it into the scenario, or you should register some kind of global shutdown function that cleans it up (e.g. atexit)